students/eowyn/session04/trigrams.py

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env python
+
+
+import random
+import string
+
+# pg 125 of think python has line processing example
+PUNCTUATION = ['.', '!', '?']
+
+# TO DO: figure out how to properly capitalize I, proper nouns
+# TO DO: Fix trigrams that end in words like "and", prepositional phrases, articles
+
+
+def convert_text(filename):
+    ''' 
+    Convert the full text of a local filename (str) to 
+    return the corresponding dictionary of trigrams
+    '''
+    words = []
+    for lines in open(filename):
+        words += process_line(lines)
+    trigrams = {}  # a dictionary
+    for i in range(len(words) - 2):
+        # doesn't include the last char, so this is a pair of words
+        pair = tuple(words[i:i + 2])
+        follower = words[i + 2]
+        trigrams.setdefault(pair,[]).append(follower)
+    return trigrams
+
+
+def construct_text(filename, n):
+    ''' Given a filename (str) and number, n, construct a 
+    made-up text of n random trigrams from the text contained in filename.
+    n should be an int and will otherwise be rounded to int.
+    '''
+    n = round(n)
+    trigrams = convert_text(filename)  # dictionary
+    new_words = []
+    for triplet in range(n):
+        # randomly select a LIST key from the dictionary
+        randokey = random.choice(list(trigrams))
+        # randomly select a STRING value from the pre-selected key
+        randoval = random.choice(trigrams[randokey])
+        new_words += [randokey[0]] + [randokey[1]] + \
```

Comment:
this is a bit klunky -- making a bunch of one element lists so you can add them. I'd probably do:

`new_words += [randokey[0], randokey[1], randoval]`